pkg/utils/subscription.go

```diff
@@ -978,3 +981,29 @@ func IsSubscriptionBeDeleted(clt client.Client, subKey types.NamespacedName) boo
 
 	return !subIns.GetDeletionTimestamp().IsZero()
 }
+
+// IsHub determines the hub cluster by listing multiclusterhubs resource items
+func IsHub(config *rest.Config) bool {
+	var dl dynamic.ResourceInterface
+
+	multiclusterHubGVR := schema.GroupVersionResource{
+		Group:    "operator.open-cluster-management.io",
+		Version:  "v1",
+		Resource: "multiclusterhubs",
+	}
+
+	dynamicClient := dynamic.NewForConfigOrDie(config)
+
+	dl = dynamicClient.Resource(multiclusterHubGVR)
+
+	objlist, err := dl.List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Infof("Listing multiclusterHub resource failed, err: %v", err)
+		return false
```

Comment:
Xiangjing I think this return false is dangerous. Because if the apiserver is down for a bit then even if its a hub, it will say its not a hub and we are back to 2 helm controllers again. I think we should error out and let it crash in this case. What do you think?

Reply:
I think we can look if there is the mch api/crd exists first if it doesn't exists then its for sure not a hub. But if it does then we proceed with the list which makes the list not fail on errors like no such api/crd. What do you think?